chaco/image_plot.py

```diff
@@ -43,7 +44,7 @@
 KIVA_DEPTH_MAP = {3: "rgb24", 4: "rgba32"}
 
 
-class ImagePlot(Base2DPlot):
+class ImagePlot(Base2DPlot, Handler):
```

Comment:
This shouldn't inherit from `Handler`.  This creates a problem with the life-cycle of the executor - we may need Enable to send a new type of event to indicate that the plot is being closed.

Reply:
How about overriding the `cleanup` method of `Component` and adding codes that stop the executor there?